synapse/storage/databases/main/appservice.py

```diff
@@ -369,17 +369,22 @@ def get_new_events_for_appservice_txn(txn):
     async def get_type_stream_id_for_appservice(
         self, service: ApplicationService, type: str
     ) -> int:
+        if type not in ["read_receipt", "presence"]:
+            raise ValueError("type must be a valid application stream id type")
```

Comment:
Can you:

1. Include the invalid type in the error message.
2. Use a `tuple` instead of a `list` (parentheses instead of brackets).